packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx

```diff
@@ -1241,6 +1251,46 @@ export class ComboBox extends BaseComponent<IComboBoxProps, IComboBoxState> {
     }
   }
 
+  private _notifyPreviewChanged(prevState: IComboBoxState) {
+    const { onPreviewExecute, onRevertPreviewExecute } = this.props;
+    const {
+      currentOptions,
+      currentPendingValueValidIndex,
+      currentPendingValueValidIndexOnHover
+    } = this.state;
+
+    if (onPreviewExecute && onRevertPreviewExecute) {
+      let sendRevert = false;
+      let currentPendingIndex = -1;
+
+      if (currentPendingValueValidIndexOnHover !== prevState.currentPendingValueValidIndexOnHover) {
+        if (this._indexWithinBounds(currentOptions, currentPendingValueValidIndexOnHover)) {
+          currentPendingIndex = currentPendingValueValidIndexOnHover;
+        } else {
+          sendRevert = true;
+        }
+      }
+
+      if (currentPendingIndex < 0 && currentPendingValueValidIndex !== prevState.currentPendingValueValidIndex) {
+        if (this._indexWithinBounds(currentOptions, currentPendingValueValidIndex)) {
+          currentPendingIndex = currentPendingValueValidIndex;
+        } else {
+          sendRevert = true;
+        }
+      }
+
+      if (this._isInPreview && (sendRevert || currentPendingIndex >= 0)) {
+        onRevertPreviewExecute();
+        this._isInPreview = false;
+      }
+
+      if (!this._isInPreview && currentPendingIndex >= 0) {
+        onPreviewExecute(currentOptions[currentPendingIndex], currentPendingIndex);
+        this._isInPreview = true;
+      }
+    }
```

Comment:
What about the case when a user enters a value that doesn't already exist as an option on the comboBox (when allowFreeform is true)

Reply:
That case I'm not sure if it's handled in Excel, so I'm skipping it. In legacy ribbon only available menu items have live preview.